### PR TITLE
Ensure 10 consecutive curls get http 200 response before saying the s…

### PR DESCRIPTION
**Pull request by hdushan**
Ensure 10 consecutive curls get http 200 response before saying the application is UP. This is to ensure that all haproxy's have been updated to point to the new service.

### CODE REVIEW

**Review by aarongorka — APPROVED** [review]
This is great @hdushan!

**Review by adenot — APPROVED** [review]

File: scripts/rancher_check_if_app_is_up.sh [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
+LONG_SLEEP=10
+FAILURE_COUNT=0
+SUCCESS_COUNT=0
+MIN_SUCCESS_COUNT=10
 
-  COUNT=$[$COUNT + 1]
-  if [ $COUNT -gt 90 ]; then
+while [ $SUCCESS_COUNT -lt $MIN_SUCCESS_COUNT ]; do
+
+  if [ $FAILURE_COUNT -gt 90 ]; then
```
Comment (theblindfrog) [inline scripts/rancher_check_if_app_is_up.sh]: Could we extract the failure count to a var? Maybe helpful for others with different requirements? Might be a tad pedantic though..

**Comment by theblindfrog** [comment]
@hdushan this is awesome!

**Comment by theblindfrog** [comment]
@hdushan @adenot @aarongorka - this change in `energy-funnel` failed.
GoCD logs here: https://gocd.amaysim.net/go/tab/build/detail/energy-funnel/473/deployToDev/1/deploy

**Pull request merged**